src/utils/querybuilder.util.js

```diff
@@ -514,16 +515,49 @@ const quantityQueryBuilder = function ({ target, field }) {
                 qB[`${field}.value`] = { $gte: num };
                 break;
             case 'ne':
-                qB[`${field}.value`] = { $ne: num };
+                offset = calcRange(num);
+                num = Number(num);
+                qB[`${field}.value`] = { $ne: { $gt: num - offset, $lt: num + offset }};
                 break;
         }
     } else {
         // no prefixes
-        qB[`${field}.value`] = { $gt: 0.99 * num, $lt: 1.01 * num };
+        offset = calcRange(num);
+        num = Number(num);
+        qB[`${field}.value`] = { $gt: num - offset, $lt: num + offset };
```

Comment:
in case of (num +- offset) precision needs to be handled
For example 
1. in case of "100.001"
    Currently the range is {'$gt':'100.0005','$lt':'100.00150000000001'}
    This will also include results containing 100.0015 that will round off to 100.002 instead of 100.001. 
    Expected range is {'$gte':'100.0005','$lt':'100.0015'}

2. in case of "100.01"
    Currently the range is {'$gt':'100.00500000000001','$lt':'100.015'}
    This will exclude 100.005 from result which should have been included. 
    Expected range is {'$gte':'100.005','$lt':'100.015'}

Reply:
Fixed range precision